packages/app/src/cli/services/dev/app-events/app-watcher-esbuild.ts

```diff
@@ -65,6 +67,22 @@ export class ESBuildContextManager {
     await Promise.all(promises)
   }
 
+  async rebuildContext(extension: ExtensionInstance) {
+    const context = this.contexts[extension.handle]
+    if (!context) return
+    await context.rebuild()
+
+    // The default output path for a extension is now inside `.shopify/bundle/<ext_id>/dist`,
+    // all extensions output need to be under the same directory so that it can all be zipped together.
+
+    // But historically the output was inside each extension's directory.
+    // To avoid breaking flows that depend on this, we copy the output to the old location.
+    // This also makes it easier to access sourcemaps or other built artifacts.
+    const outputPath = dirname(extension.getOutputPathForDirectory(this.outputPath))
+    const copyPath = dirname(extension.outputPath)
+    await copyFile(outputPath, copyPath)
```

Comment:
Checking: This is copying the directory, right? Not just the bundled file? So sourcemaps / manifests would get copied too?

Reply:
correct, this "copyFile" function also copies full directories